### ISSUE TITLE
[java] Support JDK 16

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#3079]
[OpenJDK 16](http://openjdk.java.net/projects/jdk/16/) GA is scheduled for 2021/03/16. So we should implement
support with PMD 6.32.0

The following JEPs are important for us:

*   JEP 394: [Pattern Matching for instanceof](https://openjdk.java.net/jeps/394)
    *   Was in JDK14+JDK15 a preview feature
    *   In JDK16 it will be a standard feature
*   JEP 395: [Records](https://openjdk.java.net/jeps/395)
    *   Was in JDK14+JDK15 a preview feature
    *   In JDK16 it will be a standard feature
*   JEP 397: [Sealed Classes (Second Preview)](https://openjdk.java.net/jeps/397)

In summary: https://javaalmanac.io/jdk/16/

This issue is similar to the previous: #2646

EA Release of JDK16 is available here: http://jdk.java.net/16/

Tasks: see #3111
